app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -44,9 +60,36 @@ class PopupView : FrameLayout {
     var popup: Popup? = null
         set(value) {
             field = value
-            popupAttributeListAdapter.submitList(getPopupAttributeList())
+            popupAttributeListAdapter.submitList(popupManager.displayedFields)
         }
 
+    var editMode: Boolean = false
+        set(value) {
+            field = value
+            if (field) {
+                popupAttributeListAdapter.submitList(popupManager.editableFields)
+                popupManager.startEditing()
+            } else if (popup != null) {
```

Comment:
Is this null check required here? Wouldn't we want to show display fields regardless of popup being null?

Reply:
no longer relevant